#!/usr/bin/env python

PACKAGE='object_tracking'
import roslib; roslib.load_manifest(PACKAGE)

from driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add( "predict_variance", double_t, SensorLevels.RECONFIGURE_RUNNING, "Variance for a diagonal control covariance matrix.", 1, 0.000001,  100 )
gen.add( "update_variance", double_t, SensorLevels.RECONFIGURE_RUNNING, "Variance for a diagonal update covariance matrix.", 1, 0.000001,  100 )
gen.add( "initial_variance", double_t, SensorLevels.RECONFIGURE_RUNNING, "Variance for a diagonal state covariance matrix.", 1, 0.000001,  100000 )
gen.add( "kill_var", double_t, SensorLevels.RECONFIGURE_RUNNING, "Kill kalman filters if their covariance determinant exceeds this threshold", 10e18, 1, 10e35)

gen.add( "pass_var", double_t, SensorLevels.RECONFIGURE_RUNNING, "Publish kalman filters if their covariance determinant is below this threshold", 10e17, 1, 10e35)

exit(gen.generate(PACKAGE, "object_tracker", "ObjectTracker"))